packages/next/client/image.tsx

```diff
@@ -723,29 +784,79 @@ export default function Image({
           ) : null}
         </span>
       ) : null}
+      <ImageElement raw={false} {...imgElementArgs} />
+      {priority ? (
+        // Note how we omit the `href` attribute, as it would only be relevant
+        // for browsers that do not support `imagesrcset`, and in those cases
+        // it would likely cause the incorrect image to be preloaded.
+        //
+        // https://html.spec.whatwg.org/multipage/semantics.html#attr-link-imagesrcset
+        <Head>
+          <link
+            key={
+              '__nimg-' +
+              imgAttributes.src +
+              imgAttributes.srcSet +
+              imgAttributes.sizes
+            }
+            rel="preload"
+            as="image"
+            href={imgAttributes.srcSet ? undefined : imgAttributes.src}
+            {...linkProps}
+          />
+        </Head>
+      ) : null}
```

Comment:
Why remove the priority on `raw` layout?

There's  a difference between:
- `raw` as in "no NextJs responsive layout so you have control on the styling"
- `raw` as in "it's a barebone component, you do everything yourself"

The way `raw` was described is the former, not the later.

Thus `raw` should benefit from `priority` as you want to remove NextJs styling, not remove the optimization you'd get from preloading if you pass the prop.

Reply:
Thanks for pointing this out--it was just an oversight on my part. The intention was always for `raw` to work with `priority`. I've fixed this and added a test for it.